### ISSUE TITLE
billboard being incorrectly culled

### ISSUE THREAD

**Issue by mramato** [AnalyticalGraphicsInc/cesium#4510]
1. Run the below code.
2. Gecoder to `Matagorda Island` and double click on the only placemark in the upper right (also `Matagorda Island`)
3. Rotate the camera up and down and the billboard will blink on and off.

```javascript
var viewer = new Cesium.Viewer('cesiumContainer', {
    baseLayerPicker: false,
    terrainProvider: new Cesium.CesiumTerrainProvider({
        url : 'https://assets.agi.com/stk-terrain/world',
        requestWaterMask : true,
        requestVertexNormals : true
    })
});
var options = {
    camera : viewer.scene.camera,
    canvas : viewer.scene.canvas,
    clampToGround: true
};
viewer.dataSources.add(Cesium.KmlDataSource.load('../../SampleData/kml/facilities/facilities.kml', options));
```

![bug](https://cloud.githubusercontent.com/assets/1495196/19610827/7de2d2aa-97ac-11e6-9258-cad20c3d2fd1.gif)

**Comment by pjcozzi** [comment]
Looks like a multifrustum issue.  I think there is a duplicate issue, but I couldn't find it now.

**Comment by mmccartn** [comment]
I can reproduce this in the latest release 1.26

**Comment by pjcozzi** [comment]
From #2139 via @shunter 

> First noticed this with NORAD Tracks Santa, but saw it again today.

> Billboards appear to be clipped against the edge of the frustum.

> ![image](https://cloud.githubusercontent.com/assets/32828/4342160/803f0f10-4041-11e4-8db5-a9347b7c1adc.png)

> Sandcastle example:

> https://gist.github.com/shunter/ac663bf2b527bac7242b

**Comment by emackey** [comment]
In the above screenshot, I suspect *all* of those yellow points are actually sticking into the globe.  Cesium does a good job keeping things like billboards from "sticking into the globe", but the globe-depth tricks break down across frustum boundaries.  So for the clipped yellow dots above, the yellow fragments are being rendered into a rear frustum, and the globe imagery is being rendered into a closer frustum, and there's a depth buffer clear in between.  There's no easy way to allow the points to show through the globe at the boundary, because they are legitimately occluded.

**Comment by pjcozzi** [comment]
Also reported in #6152

**Comment by laurensdijkstra** [comment]
In #6152 I set `disableDepthTestDistance: Infinity`, so in theory the Billboard should in that case not be depth tested against the terrain. But I guess the Billboard then ends up in front of all the Primitives (including the Globe) a background Frustum, while the Frustum in front of it is just drawn over it, as it should always be in front of the previous Frustum. But this conflicts with the depth testing that is disabled on the Billboard.

**Comment by emackey** [comment]
> So perhaps every Primitive with depth testing disabled should end up in the front most Frustum.

My current thinking on this topic is that all 2D primitives end up in their own render pass, with some kind of single-3D-point depth test to determine visibility status of the primitive.  The primitive would exist outside of any 3D frustums and would not have per-fragment depth testing.  See thread: https://github.com/AnalyticalGraphicsInc/cesium/issues/2694#issuecomment-338168783

**Comment by pjcozzi** [comment]
@emackey @laurensdijkstra yes, has potential - may still z fight in the distance - will need to evaluate - might not be bad with #5851.

**Comment by ggetz** [comment]
Reported on the forum: https://groups.google.com/forum/#!topic/cesium-dev/odobrazRIWU
